### PR TITLE
examples: fix flag syntax for zkctl

### LINKED ISSUE #14471: Bug Report: local example fails to run with zookeeper topo
### Overview of the Issue

Running local example with zookeeper produces errors.
```
Waiting for zk servers to be ready...
ZK server number 1 failed to start. See log:
    /Users/deepthi/go/src/vitess.io/vitess/vtdataroot/tmp/zkctl_1.out
ZK server number 2 failed to start. See log:
    /Users/deepthi/go/src/vitess.io/vitess/vtdataroot/tmp/zkctl_2.out
ZK server number 3 failed to start. See log:
    /Users/deepthi/go/src/vitess.io/vitess/vtdataroot/tmp/zkctl_3.out
```

### Reproduction Steps

Run the local example like this:

```sh
$ TOPO=zk2 ./101_initial_cluster.sh
```

### Binary Version

```sh
v16.0.0+
```

### Operating System and Environment details

```sh
Any
```

### Log Fragments

```sh
ERROR: logging before flag.Parse: E1105 13:48:16.323815    2963 zkctl.go:29] unknown command "1" for "zkctl"
```

### PR DESCRIPTION
**Pull request by deepthi**
## Description
Running the local example with `TOPO=zk2` fails with an error from zkctl startup.
This is because we changed flags to use `--` instead of `-` back in v16.
This was broken at least on main, most likely on 17, and probably on v16 as well.

## Related Issue(s)
Fixes #14471 

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on the CI
-   [x] Documentation was added or is not required

### CODE REVIEW

**Comment by deepthi** [comment]
We don't see to run local example with zookeeper in CI. If we did, we might have caught this.
OTOH, we still have this, though we've deleted the k8s topo. I'm not sure how that is passing.
```
local_example / Local example using k8s on ubuntu-22.04 (pull_request) 
```

**Comment by dbussink** [comment]
> OTOH, we still have this, though we've deleted the k8s topo. I'm not sure how that is passing.

Code looks like this:

```
# start topo server
if [ "${TOPO}" = "zk2" ]; then
  CELL=zone1 ../common/scripts/zk-up.sh
elif [ "${TOPO}" = "consul" ]; then
  CELL=zone1 ../common/scripts/consul-up.sh
else
  CELL=zone1 ../common/scripts/etcd-up.sh
fi
```

So it considers the default to be `etcd`, if you set something random / unknown.

**Review by dbussink — APPROVED** [review]

File: .github/workflows/local_example.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ jobs:
     runs-on: gh-hosted-runners-16cores-1
     strategy:
       matrix:
-        topo: [consul,etcd,k8s]
+        topo: [consul,etcd,zk2]
```
Comment (mattlord) [inline .github/workflows/local_example.yml:11]: Ah, that’s why we didn’t notice 🙂

**Review by mattlord — APPROVED** [review]

**Comment by mattlord** [comment]
Looks like we didn’t remove the k8s local example workflow before either.

File: .github/workflows/local_example.yml (hunk 1 above)
Comment (mattlord) [reply .github/workflows/local_example.yml:11]: I removed the local k8s example from the required checks for main.

Comment (mattlord) [reply .github/workflows/local_example.yml:11]: Maybe we want to make the zk2 one required? Wasn’t sure about that.

**Pull request merged**